libsignal-service-actix/src/push_service.rs

```diff
@@ -182,6 +183,70 @@ impl PushService for AwcPushService {
         ))
     }
 
+    async fn post_to_cdn0<'s, C: std::io::Read + Send + 's>(
+        &mut self,
+        path: &str,
+        value: &[(&str, &str)],
+        file: Option<(&str, &'s mut C)>,
+    ) -> Result<(), ServiceError> {
+        let url = Url::parse(&self.cfg.cdn_urls[&0])
+            .expect("valid cdn base url")
```

Comment:
I guess now that we use `thiserror` everywhere, it would be nicer to have the `uri::ParseError` mapped there instead of unwrapping?

Reply:
Should we not test this in the constructor instead?

Reply:
Of `ServiceConfiguration`? Well, yes, but... catch all errors and never rely on the implementation of another struct :crab:

Reply:
-> https://github.com/Michael-F-Bryan/libsignal-service-rs/issues/48